src/plugins/intel_cpu/src/nodes/rdft.cpp

```diff
@@ -0,0 +1,939 @@
+    }
+
+    const auto& axesPrecision = getOriginalInputPrecisionAtPort(AXES_INDEX);
+    if (axesPrecision != Precision::I32 && axesPrecision != Precision::I64) {
+        IE_THROW() << errorMsgPrefix << " has unsupported 'axes' input precision: " << axesPrecision.name();
+    }
+
+    if (inputShapes.size() > SIGNAL_SIZE_INDEX) {
+        const auto& signalSizePrecision = getOriginalInputPrecisionAtPort(SIGNAL_SIZE_INDEX);
+        if (signalSizePrecision != Precision::I32 && signalSizePrecision != Precision::I64) {
+            IE_THROW() << errorMsgPrefix << " has unsupported 'signalSize' input precision: " << signalSizePrecision.name();
+        }
+    }
+
+    std::vector<PortConfigurator> configurators({{LayoutType::ncsp, Precision::FP32},
+                                                 {LayoutType::ncsp, Precision::I32}});
+    if (inputShapes.size() > SIGNAL_SIZE_INDEX)
+        configurators.push_back({LayoutType::ncsp, Precision::I32});
+
+    addSupportedPrimDesc(configurators, {{LayoutType::ncsp, Precision::FP32}}, impl_desc_type::ref_any);
+}
+
+void RDFT::execute(dnnl::stream strm) {
+    const auto& inputMem = getParentEdgeAt(DATA_INDEX)->getMemory();
+    const auto& outputMem = getChildEdgeAt(0)->getMemory();
+    const auto& inputShape = inputMem.getStaticDims();
+    const auto& outputShape = outputMem.getStaticDims();
+
+    auto inputPtr = reinterpret_cast<float*>(inputMem.GetPtr());
+    auto outputPtr = reinterpret_cast<float*>(outputMem.GetPtr());
+
+    auto rank = inputShape.size() - inverse;
+
+    if (axes.size() == 0) {
+        const auto& axesMem = getParentEdgeAt(AXES_INDEX)->getMemoryPtr();
+        auto axesPtr = reinterpret_cast<const int32_t*>(axesMem->GetPtr());
+        axes = std::vector<int>(axesPtr, axesPtr + axesMem->getStaticDims()[0]);
+        normalizeAxes(axes, rank);
+    }
+
+    if (signalSizes.size() == 0) {
+        if (SIGNAL_SIZE_INDEX < getOriginalInputsNumber()) {
+            const auto& signalSizeMem = getParentEdgeAt(SIGNAL_SIZE_INDEX)->getMemoryPtr();
+            auto signalPtr = reinterpret_cast<const int32_t*>(signalSizeMem->GetPtr());
+            signalSizes = std::vector<int>(signalPtr, signalPtr + signalSizeMem->getStaticDims()[0]);
+        } else {
+            signalSizes = getDefaultSignalSizes(inputShape, axes, inverse);
+        }
+    }
+
+    const auto& inputStrides = inputMem.GetDescWithType<BlockedMemoryDesc>()->getStrides();
+    const auto& outputStrides = outputMem.GetDescWithType<BlockedMemoryDesc>()->getStrides();
+
+    if (twiddles.size() == 0) {
+        twiddles = executor->generateTwiddles(signalSizes, outputShape, axes);
+    }
+
+    executor->execute(inputPtr, outputPtr,
+                      twiddles, rank,
+                      axes, signalSizes,
+                      inputShape, outputShape,
+                      inputStrides, outputStrides);
+}
+
+bool RDFT::created() const {
+    return getType() == Type::RDFT;
+}
+
```

Comment:
Why did you override the createPrimitive and needPrepareParams functions?

Reply:
right, it's redundant